File: Main/static/JS/game/effects/bleeding.js
import { DurableEffect, Effect } from "./effect.js";


class Bleeding extends Effect {
    constructor(hero, game_state, caster, value){
        super('кровотечение', 'bleeding.png', hero, game_state, caster)
        this.value = value
    }

    gen_description(){
        return `Кровотечение [${this.value}] (наложено ${this.caster.name})`
    }

    on_getting(){
        var sum_value = 0
        for (let effect of this.hero.effects){
            if (effect.name=='кровотечение'){
                sum_value += effect.value
                effect.is_alive = false
            }
        }
        
        this.is_alive = true
        this.value = sum_value
    }

    decrease(){
        this.value -= 1
        if (this.value == 0){
            this.is_alive = false
        }
    }

    before_target_move(){
        this.game_state.log(`${this.hero.name} кровоистекает. Он потерял ${this.value} hp`)
        this.hero.loose_hp(this.value, this.caster)
        this.decrease()
        this.hero.be_attacked_animation()
    }
}


export {Bleeding}

File: Main/models/effects/bleeding.py
from .effect import SummaryDurableEffect


class Bleeding(SummaryDurableEffect):
    name = 'Bleeding'
    img_src = 'bleeding.png'

    def gen_description(self, link):
        return f'Кровотечение[{self.duration}] (наложено {link.caster.name})'

    def before_target_move(self, link):
        game_state = link.hero.game_state
        link.hero.loose_hp(self.duration)
        game_state.create_ui_action('damage', link.hero.id, 0.2)
        self.decrease_duration(link)
